src/zimcheck.cpp

```diff
@@ -92,27 +92,29 @@ static std::unordered_map<TestType, std::pair<LogTag, std::string>> errormapping
 class ErrorLogger {
   private:
     std::unordered_map<TestType, std::vector<std::string>> errors;
-    std::unordered_map<TestType, bool> testStatus;
+    std::unordered_map<TestType, std::pair<bool, bool>> testStatus;
 
   public:
     ErrorLogger()
     {
         for (const auto &m : errormapping) {
-            testStatus[m.first] = true;
+            testStatus[m.first].first = true;
+            testStatus[m.first].second = false;
         }
     }
 
     void setTestResult(TestType type, bool status) {
-        testStatus[type] = status;
+        testStatus[type].first = status;
     }
 
     void addError(TestType type, const std::string& message) {
+        testStatus[type].second = true;
```

Comment:
Could we have the `addError` method also set the test result ?
If we add error message, we know that the test is failing.

Reply:
I renamed to addReportMsg() it makes much more sense for me. Let me know what you think